datadog-integrations-aws-handler/src/datadog_integrations_aws/handlers.py

```diff
@@ -0,0 +1,191 @@
+
+    aws_account = build_aws_account_from_model(model)
+    with v1_client(
+            model.DatadogCredentials.ApiKey,
+            model.DatadogCredentials.ApplicationKey,
+            model.DatadogCredentials.ApiURL or "https://api.datadoghq.com",
+            TELEMETRY_TYPE_NAME,
+            __version__,
+    ) as api_client:
+        api_instance = AWSIntegrationApi(api_client)
+        try:
+            api_instance.update_aws_account(
+                aws_account,
+                account_id=model.AccountID,
+                role_name=model.RoleName,
+                access_key_id=model.AccessKeyID,
+            )
+        except ApiException as e:
+            LOG.error("Exception when calling AWSIntegrationApi->update_aws_account: %s\n", e)
+            return ProgressEvent(
+                status=OperationStatus.FAILED, resourceModel=model, message=f"Error updating AWS account: {e}"
+            )
+
+    return read_handler(session, request, callback_context)
+
+
+@resource.handler(Action.DELETE)
+def delete_handler(
+        session: Optional[SessionProxy],
+        request: ResourceHandlerRequest,
+        callback_context: MutableMapping[str, Any],
+) -> ProgressEvent:
+    model = request.desiredResourceState
+
+    aws_account = build_aws_account_from_model(model)
+
+    with v1_client(
+            model.DatadogCredentials.ApiKey,
+            model.DatadogCredentials.ApplicationKey,
+            model.DatadogCredentials.ApiURL or "https://api.datadoghq.com",
+            TELEMETRY_TYPE_NAME,
+            __version__,
+    ) as api_client:
+        api_instance = AWSIntegrationApi(api_client)
+        try:
+            api_instance.delete_aws_account(aws_account)
+        except ApiException as e:
+            LOG.error("Exception when calling AWSIntegrationApi->delete_aws_account: %s\n", e)
+            return ProgressEvent(
+                status=OperationStatus.FAILED, resourceModel=model, message=f"Error deleting AWS account: {e}"
+            )
+
+    return ProgressEvent(
+        status=OperationStatus.SUCCESS,
+        resourceModel=model,
+    )
+
+
+@resource.handler(Action.READ)
+def read_handler(
+        session: Optional[SessionProxy],
+        request: ResourceHandlerRequest,
+        callback_context: MutableMapping[str, Any],
+) -> ProgressEvent:
+    model = request.desiredResourceState
+
+    with v1_client(
+            model.DatadogCredentials.ApiKey,
+            model.DatadogCredentials.ApplicationKey,
+            model.DatadogCredentials.ApiURL or "https://api.datadoghq.com",
+            TELEMETRY_TYPE_NAME,
+            __version__,
+    ) as api_client:
+        api_instance = AWSIntegrationApi(api_client)
+        try:
+            aws_account = api_instance.list_aws_accounts(
+                account_id=model.AccountID,
+                role_name=model.RoleName,
+                access_key_id=model.AccessKeyID
+            ).accounts[0]
+        except ApiException as e:
```

Comment:
Should we also catch an index error here in case the account isn't in the list? (Not sure if thats possible, or if you'd get the ApiException instead)